PIMbench/cpp-relu/PIM/relu.cpp

```diff
@@ -259,72 +254,74 @@ int main(int argc, char *argv[]) {
   int inputWidth = inputMatrix[0][0].size();
 
   // Define parameters for processing
-  unsigned numCols = 8192, numOfCore = 4096;
-  int numOfPIMRow = 1;
-  int numOfPIMColumn = (inputHeight * inputWidth / numOfPIMRow);
-  int numOfMatPerRow = std::min(static_cast<int>(std::floor((1.0 * numCols * numOfCore) / numOfPIMColumn)), inputDepth);
+  PimDeviceProperties deviceProp;
+  PimStatus status = pimGetDeviceProperties(&deviceProp);
+  if (status != PIM_OK) {
+    std::cout << "Abort: pimGetDeviceProperties failed" << std::endl;
+    return 1;
+  }
+  // Get the device parameters
+  uint64_t numCols = deviceProp.numColPerSubarray;
+  uint64_t numRows = deviceProp.numRowPerSubarray;
+  uint64_t numOfBits = uint64_t(deviceProp.numRanks) * uint64_t(deviceProp.numBankPerRank) * uint64_t(deviceProp.numSubarrayPerBank) * numCols * numRows;
+
+  uint64_t numOfPIMRow = 1;
+  uint64_t numOfMatPerRow = std::min(static_cast<uint64_t>(std::floor((1.0 * numOfBits) / (inputHeight * inputWidth * 32))), static_cast<uint64_t>(inputWidth));
 
   // Initialize result matrix with the same dimensions as input matrix
   std::vector<std::vector<std::vector<int>>> resultMatrix(inputDepth, std::vector<std::vector<int>>(inputHeight, std::vector<int>(inputWidth)));
 
-  int tempcol = 0;
   std::vector<std::vector<int>> decompMat;
   std::vector<std::vector<int>> mergedMat(numOfPIMRow);
   std::vector<int> outVector;
   outVector.resize(inputDepth * inputHeight * inputWidth);
   
   // Loop through input depth in chunks
-  for (int j = 0; j < inputDepth; j += numOfMatPerRow) {
-    int matChunk = (numOfMatPerRow + j) <= inputDepth ? (numOfMatPerRow + j) : inputDepth;
+  for (uint64_t j = 0; j < inputDepth; j += numOfMatPerRow) {
+    uint64_t matChunk = (numOfMatPerRow + j) <= inputDepth ? (numOfMatPerRow + j) : inputDepth;
     // Decompose and merge matrices
-    for (int k = j; k < matChunk; k++) {
-      decomposeMatrix(params.row, params.column, 1, 1, 1, 0, inputMatrix[k], decompMat);
+    for (uint64_t k = j; k < matChunk; k++) {
+      decomposeMatrix(inputHeight, inputWidth, 1, 1, 1, 0, inputMatrix[k], decompMat);
```

Comment:
Can you please add some comments on why these values are 1, 1, 1, 0

Reply:
Added comments and updated the file.